src/modules/inspect-correlation/components/token-viewer/token-viewer.ts

```diff
@@ -22,60 +22,133 @@ export class TokenViewer {
-        };
-
-        const historyEntryHasPayload: boolean = historyEntry.payload !== undefined;
-        if (historyEntryHasPayload) {
-          for (const load in historyEntry.payload) {
-            const payloadEntry: IPayLoadEntry = {
-              name: load,
-              values: [],
-            };
-
-            for (const entry in historyEntry.payload[load]) {
-              payloadEntry.values.push({
-                title: entry,
-                value: JSON.stringify(historyEntry.payload[load][entry]),
-              });
-            }
-
-            tokenEntry.payload.push(payloadEntry);
-          }
-        }
-
-        this.tokenEntries.push(tokenEntry);
-      });
+    return formattedTokenEntryPayload;
+  }
+
+  private _getAllPayloadEntriesForObject(payload: any): Array<IPayloadEntry> {
+    const payloadEntries: Array<IPayloadEntry> = [];
 
-      this.showTokenEntries = true;
-    } catch (error) {
-      this.showTokenEntries = false;
+    for (const loadIndex in payload) {
+      const currentLoad: any = payload[loadIndex];
+
+      const payloadEntry: IPayloadEntry = this._getPayloadEntryForObject(currentLoad, loadIndex);
+
+      payloadEntries.push(payloadEntry);
     }
 
+    return payloadEntries;
+  }
+
+  private _getPayloadEntryForObject(load: any, loadName: string): IPayloadEntry {
+    const payloadEntry: IPayloadEntry = {
+      name: loadName,
+      values: [],
+    };
+
+    const entryIsNotAnObject: boolean = typeof load !== 'object';
+    if (entryIsNotAnObject) {
+      const payloadEntryValues: Array<IPayloadEntryValue> = this._getPayloadEntryValuesForNonObject(load);
+
+      payloadEntry.values = payloadEntryValues;
+    } else {
+      const payloadEntryValues: Array<IPayloadEntryValue> = this._getPayloadEntryValuesForObject(load);
+
+      payloadEntry.values = payloadEntryValues;
+    }
+
+    return payloadEntry;
+  }
+
+  private _getPayloadEntryValuesForObject(payload: any): Array<IPayloadEntryValue> {
+    const payloadEntryValues: Array<IPayloadEntryValue> = [];
+
+    for (const entryIndex in payload) {
+      // tslint:disable-next-line no-magic-numbers
+      const payloadEntryValue: string = JSON.stringify(payload[entryIndex], null, 2);
+
+      payloadEntryValues.push({
+        title: entryIndex,
+        value:  payloadEntryValue,
+      });
+    }
+
+    return payloadEntryValues;
+  }
+
+  private _getPayloadEntryForNonObject(payload: any): IPayloadEntry {
+    const payloadEntryValues: any = this._getPayloadEntryValuesForNonObject(payload);
+
+    const payloadEntry: IPayloadEntry = {
+      values: payloadEntryValues,
+    };
+
+    return payloadEntry;
+  }
+
+  private _getPayloadEntryValuesForNonObject(payload: any): Array<IPayloadEntryValue> {
+    const payloadIsString: boolean = typeof payload === 'string';
+
+    const payloadEntryValue: string = payloadIsString
+                                  ? `"${payload}"`
+                                  : payload;
```

Comment:
When `payload` is not a string, it will be assigned to a `string` field. To be safe, I would call `toString()` here.

```suggestion
                                  : payload.toString();
```